.storybook/src/zoom.stories.ts

```diff
@@ -0,0 +1,285 @@
+import * as d3 from 'd3';
+import CanvasLayer from '../../src/layers/CanvasLayer';
+import {
+  OnMountEvent,
+  OnUpdateEvent,
+  OnRescaleEvent,
+  LayerOptions,
+} from '../../src/interfaces';
+import GridLayer from '../../src/layers/GridLayer';
+import ZoomPanHandler from '../../src/control/ZoomPanHandler';
+import { scaleLinear, curveStep } from 'd3';
+
+const width = 600;
+const height = 400;
+
+function createButton(label: string, cb: any){
+  const btn = document.createElement('button');
+  btn.innerHTML = label;
+  btn.setAttribute('style', 'width: 100px;height:32px;margin-top:12px;')
+  btn.onclick = cb;
+  return btn;
+}
+
+class TestLayer extends CanvasLayer {
+  constructor(id: String, options: LayerOptions) {
+    super(id, options);
+
+    this.render = this.render.bind(this);
+  }
+
+  onRescale(event: OnRescaleEvent) {
+    super.onRescale(event);
+
+    this.render(event);
+  }
+
+  onUpdate(event: OnUpdateEvent) {
+    super.onUpdate(event);
+
+    this.render(event);
+  }
+
+  render(event: OnRescaleEvent | OnUpdateEvent) {
+    const {
+      ctx,
+    } = this;
+
+    if (!ctx) return;
+
+    // Get the bounds area
+    const xb1 = event.xBounds[0];
```

Comment:
Could be destructed const [xb1, xb2] = event.xBounds

Reply:
Fixed